lib/linalg/svd/IncrementalSVDBrand.h

```diff
@@ -27,6 +42,20 @@ namespace CAROM {
 class IncrementalSVDBrand : public IncrementalSVD
 {
 public:
+    /**
```

Comment:
There are significant changes, especially in the cpp file, in the existing class `IncrementalSVDBrand`, to accommodate incremental DMD. Can this PR be refactored to avoid changes to `IncrementalSVDBrand`? It is understandable that the data from `IncrementalSVDBrand` needs to be accessed for incremental DMD, so it makes sense to add access functions. Hopefully it is possible to avoid changing so much of the class's implementation, by using access functions, making a new class derived from `IncrementalSVDBrand`, or some other way.

Reply:
I've somehow fixed this problem in 06b4d6b. The lengthy `IncrementalDMDInternal` things have been removed, but the thing is that we still need to intercept some of the intermediate matrices from `IncrementalSVDBrand` for `IncrementalDMD`. Specifically, matrices before `IncrementalSVDBrand::addNewSample` or `IncrementalSVDBrand::addLinearlyDependentSample` are accessed by `IncrementalDMD`. For that, I have added extra pointers pointing to the matrices before being updated. (See Line 176-190 of `lib/linalg/svd/IncrementalSVDBrand.h`. `IncrementalDMD` is also declared as a friend class of `IncrementalSVDBrand` so it can access those private pointers. I find that this makes minimal changes to `IncrementalSVDBrand`, but if you have better ideas, please let me know.

Reply:
Thanks for the improvement. There is still a lot of new code in `IncrementalSVDBrand.cpp`, and it is hard to tell whether it affects the existing incremental SVD or just adds additional intermediate matrices needed by `IncrementalDMD`. If it should not change the SVD algorithm, and if it only stores intermediate matrices, can this be done optionally by setting a flag? My concerns are (i) verifying that the old incremental SVD is not changed, and (ii) not adding extra computation or memory usage when `IncrementalDMD` is not being used.

Reply:
One way to separate the new code from the existing `IncrementalSVDBrand` might be to make a class derived from `IncrementalSVDBrand`, containing the additional data members. Functions like `buildIncrementalSVD`, `updateTemporalBasis`, and `addNewSample` could become virtual. Some code from the class `IncrementalSVDBrand` would be reused, while the new code is added only in the derived class.